`cargo-embed` → `probe-rs` documentation change

BobAdamsEE opened the pull request:
Attempting to install cargo-embed with instructions from the current `boards/atsamv71_xult/README.md` lead to the following:

```
  `cargo-embed` is now a part of `probe-rs`. To install it, run the following PowerShell command:

  irm https://github.com/probe-rs/probe-rs/releases/latest/download/probe-rs-tools-installer.ps1 | iex

  Once installed, you can use the "cargo embed" command just like before, although there might be minor changes in the API. For more information, please refer to the documentation at https://probe.rs/docs/tools/cargo-embed/.
```

Updated `boards/atsamv71_xult/README.md` with the relevant instructions for `probe-rs` for Windows / Linux.

BobAdamsEE committed:
Update README.md

`cargo-embed` is now a part of `probe-rs`.

BobAdamsEE committed:
Update CHANGELOG.md

Updating per suggestion in contribution guide

BobAdamsEE commented:
This will need further work - probe-rs has issues with the Embed.toml from changes from the initial cargo-embed. Will submit again once I get flashing working.

michalfita commented on CHANGELOG.md:
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## [Unreleased]
+- Fixed documentation in boards\atsamv71_xult to refer to probe-rs instead of cargo-embed
```
Please refrain from using backspaces in paths, this is very Windowsy, but incorrect.
```suggestion
- Fixed documentation in `boards/atsamv71_xult` to refer to `probe-rs` instead of `cargo-embed`
```

michalfita commented on boards/atsamv71_xult/README.md:
```diff
@@ -5,7 +5,12 @@ The examples are written in [RTIC](https://rtic.rs).
 ## Prerequisites
 * Install the cross-compilation toolchain: `rustup target add thumbv7em-none-eabihf`.
-* Install [cargo-embed](https://github.com/probe-rs/cargo-embed): `cargo install cargo-embed`.
+* Install [probe-rs](https://probe.rs/) (formerly cargo-embed):
```
```suggestion
* Install [`probe-rs`](https://probe.rs/) (formerly `cargo-embed`):
```

michalfita reviewed (commented):
I'm not author of this demo and I don't use it, hence missed the change that seem to happen in 2023. Thanks for efforts to improve the documentation.

tmplt commented on boards/atsamv71_xult/README.md:
```diff
@@ -5,7 +5,12 @@ The examples are written in [RTIC](https://rtic.rs).
+  
+  On Windows `irm https://github.com/probe-rs/probe-rs/releases/latest/download/probe-rs-tools-installer.ps1 | iex`
+  
+  On Linux `curl --proto '=https' --tlsv1.2 -LsSf https://github.com/probe-rs/probe-rs/releases/latest/download/probe-rs-tools-installer.sh | sh`
```
I don't like the `curl | sh` trend, even if canonical upstream. I suggest just pointing to the official probe-rs docs instead. That way we dont have to keep tabs on whether the install command changes.

BobAdamsEE replied:
I removed lines 9-12 to resolve, can refer to install methods from probe-rs website

BobAdamsEE committed:
Update Embed.toml

Fixed errors that cause cargo embed to not work including show_timestamps as a general default.rtt setting (now per up_channel), halt_afterwards in the default.flashing section, and the channels parameter of default.rtt was split into up_channels and down_channels.

BobAdamsEE committed:
Update CHANGELOG.md

Remove windows style of paths

Co-authored-by: Michał Fita <4925040+michalfita@users.noreply.github.com>

BobAdamsEE committed:
Update boards/atsamv71_xult/README.md

Co-authored-by: Michał Fita <4925040+michalfita@users.noreply.github.com>

BobAdamsEE committed:
Update README.md

Per recommendation removed curl | sh notes - users can reference current documentation from probe-rs site directly

BobAdamsEE commented:
Resolved review comments and updated Embed.TOML to handle some changes from probe-rs:

- https://github.com/probe-rs/probe-rs/pull/2419/ - halt_afterwards removed from default.flashing section
- https://github.com/probe-rs/probe-rs/pull/2249 - show_timestamps and channel parameter changes in refactor

michalfita reviewed (approved):
OK, I think it's good enough.

michalfita committed:
Merge branch 'development' into patch-1
